packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1822,16 +1825,15 @@ export class Configuration {
     }));
   }
 
-  normalizePackage(original: Package) {
+  async normalizePackage(original: Package) {
```

Comment:
I'm a bit concerned about making this asynchronous. I don't feel like the normalization itself should require I/O.

I think I'd prefer if the `packageExtensions` were a parameter from an option bag passed to `normalizePackage`, and perhaps computed when initializing the `Project` instance (or at least when calling its `install` method).

Reply:
I'm not sure I agree - semantically speaking there's nothing wrong with `normalizePackage` being async, since it requires access to resources that can only be loaded asynchronously (not necessarily doing IO).

Because of this, I believe that it should load the extensions itself - loading them in parent method just makes the code more complex for no good reason.

> and perhaps computed when initializing the Project instance

1) Computing it when initializing anything just moves the issue this PR fixes one level further.
2) I don't think the project should initialize them since they're purely tied to the configuration. Why add a layer of indirection?

Reply:
I disagree - I don't see the process of normalization depending on anything else than data available before running the function (in essence, it works by integrating a configuration into an input, transformed into an output; there's no reason why the configuration should be initialized as part of the function itself).

I'm fine with a different design than `refreshPackageExtensions`, but I don't think this is the right one.

Reply:
Fair enough, changed :+1: